2018 Robot/src/org/usfirst/frc/team4999/triggers/DriveOvercurrent.java

```diff
@@ -0,0 +1,37 @@
+package org.usfirst.frc.team4999.triggers;
+
+import org.usfirst.frc.team4999.robot.RobotMap;
+
+import edu.wpi.first.wpilibj.PowerDistributionPanel;
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj.buttons.Trigger;
+
+/**
+ *
+ */
+public class DriveOvercurrent extends Trigger {
+	
+	private static final int[] MOTORS = {0,1,2,3};
+	
+	private static final double CUTOFF_CURRENT = 30;
+	private static final double CUTOFF_TIME = 1;
+	
+	private PowerDistributionPanel pdp = RobotMap.pdp;
+	
+	private Timer time = new Timer();
+	
+	public DriveOvercurrent() {
+		super();
+		time.start();
+	}
+
+    public boolean get() {
```

Comment:
Haven't found the place this is used yet, but given this impl, it would seem you're intending to shut down everything if anything triggers.  Is that what you really want?

Reply:
The trigger is called in the OI, and it's bound to the `KillDrive` command. The scheduler polls the [Triggers](https://wpilib.screenstepslive.com/s/currentCS/m/java/l/599739-running-commands-on-joystick-input) every 20ms or so, and if `get()` returns true, it runs the associated command. In the OI, I have a button to run the `TeleopNoPID` command, which will start the robot again if the overcurrent has triggered